doc/HOWTO/howto_setupAndDeployCustom.md

```diff
@@ -0,0 +1,144 @@
+
+## Process
+
+Prerequisites:
+- Make `cargo`, `rustup`, and `go` available. You will need a`rustc` version that is at least 1.74.
+- Make sure you have enough allocated space. If you're using the NixOS VM, you might need to add `services.logind.extraConfig = "RuntimeDirectorySize=2G";` to your configuration, rebuild and reboot the VM.
+- you probably want to add `~/.cargo/bin` to your PATH
+
+### Local Stellar Network
+Instead of building from source, Stellar provides a quickstart Docker image. You can call the below command from anywhere, and Docker will fetch the image.
+```sh
+docker run --rm -d \
+    -p 8000:8000 \
+    --name stellar \
+    stellar/quickstart:testing \
+    --testnet \
+    --enable-soroban-rpc
+```
+Note here that the [official guide][guide] would have you run this in interactive mode (`run ... -it ...`), but for just the network having it run in the background (`-d`) is less noisy and doesn't hijack the terminal.
+
+The above snippet deploys a testnet (`--testnet`), if you want a standalone network, replace that flag with `--standalone`.
+
+### Soroban CLI
+In order to interact with your docker image, you will need the [Soroban CLI][cli]. With `cargo`, all you need is:
+```sh
+cargo install --locked soroban-cli --features opt
+```
+
+note that this gets placed in `~/.cargo/bin`, which is why you want that as part of your `PATH`.
+
+### Custom Soroban Contracts
+
+<!--- TODO: Make a megacontract and link it here-->
+
+You can use the Soroban CLI to set up a blank project, and add your own contract code to it (you may also optionally include any of their example contracts).
+
+To set up a new _project_ (not contract!), type:
+```sh
+soroban contract init <DIR>
+```
+This will populate `<DIR>` (e.g. `.`) with some basic files, most notably a `Cargo.toml` config. Your custom contract should be added in `<DIR>/contracts/<CONTRACT>`.
+For example, for a contract `Foo`, you'd create:
+```
+<DIR>
+├── Cargo.toml
+└── contracts
+    └── Foo
+        ├── Cargo.toml
+        └── src
+            ├── lib.rs
+            └── test.rs
+```
+
+Note that `Cargo.toml` in `Foo` is _not_ the same `Cargo.toml` in `<DIR>`! The latter gets created automagically with `soroban contract init`, the former you have to add yourself. Here's an example of `Cargo.toml` in the contract folder:
+```
+dev_dependencies = { soroban-sdk = { workspace = true } }
+[package]
+name = "foo"
+version = "0.0.0"
+edition = "2021"
+publish = false
+
+[lib]
+crate-type = ["cdylib"]
+doctest = false
+
+[dependencies]
+soroban-sdk = { workspace = true }
+
+[dev-dependencies]
+soroban-sdk = { version = "20.3.2", features = ["testutils"] }
+```
+
+Write your contract code in `lib.rs`, and tests in `test.rs`.
+Make sure all your tests work before deploying (with print outputs):
+```sh
+cargo test -- --nocapture
+```
+Once they do, running
+```sh
+soroban contract build
```

Comment:
Weird... I needed to pass `--out-dir someDir` here, otherwise it wouldn't create the `.wasm` file.

Reply:
Did you use the megacontract folder or was this a from-scratch project? And if you did use it, did you run `soroban contract init` anyway?

Reply:
I did a clean start, and now it works. Maybe I ran the `rustup target add wasm32-unknown-unknown` at the wrong point in time 🤷